### PR TITLE
BACK-1117: Bitcoin API Swagger

### PR DESCRIPTION
**Pull request by jjayet-ledger**
## What is this about?

What is says on the title, we now have a Swagger

## Cute picture of lama
lama

### CODE REVIEW

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 1]
```diff
@@ -0,0 +1,500 @@
+openapi: 3.0.1
+info:
+  title: Lama Bitcoin API
+  description: Returns beautifull curated photos
+  version: 0.3.0
+servers:
+  - url: https://TBD/
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:7]: you could setup a default value with the `http://localhost:8080`

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+paths:
+  /accounts:
+    post:
+      summary: Create a user account and trigger synchronization
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:11]: Register an account in order to trigger synchronization

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+        sync_id:
+          type: string
+          format: 'uuid'
+          example: '013260eb-7624-40d2-8327-97177a69d8f3'
+        status:
+          type: string
+          enum: [registered, unregistered, published, synchronized, sync_failed, deleted, delete_failed]
+        payload:
+          type: object
+    CreationRequest:
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:232]: RegisterAccountRequest

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+      operationId: createAccount
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreationRequest'
+      responses:
+        200:
+          description: The created user account
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:21]: Registered account

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+        network:
+          type: string
+          example: 'MainNet'
+        coin_family:
+          type: integer
+          example: 'bitcoin'
+        coin:
+          type: string
+          example: 'btc'
+    AccountCreated:
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:260]: RegisterAccountResponse

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+  /accounts/{accountId}:
+    parameters:
+      - name: accountId
+        in: path
+        required: true
+        schema:
+          type: string
+          format: 'uuid'
+    get:
+      summary: Get a user account by id
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:35]: Don't necessary to precise `user account`, just `account` is enough.
(Same for all occurences below)

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 7]
```diff
@@ -0,0 +1,500 @@
...
+          format: uuid
+          example: '013260eb-7624-40d2-8327-97177a69d8f3'
+        sync_id:
+          type: string
+          format: uuid
+          example: '013260eb-7624-40d2-8327-97177a69d8f3'
+        sync_frequency:
+          type: number
+          example: 600
+    UpdateRequest:
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:274]: UpdateAccountRequest

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 8]
```diff
@@ -0,0 +1,500 @@
...
+          required: true
+          schema:
+            type: string
+            format: 'uuid'
+        - name: limit
+          in: query
+          schema:
+            type: number
+            minimum: 1
+            maximum: 20
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:85]: Let's cap it to 1000 for maximum value (we need to apply it in the code)

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 9]
```diff
@@ -0,0 +1,500 @@
...
+      responses:
+        200:
+          description: The utxos from the account
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetUtxosResult'
+  /accounts/{accountId}/operations:
+    get:
+      summary: Get operations from an account
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:108]: Get account operations

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 10]
```diff
@@ -0,0 +1,500 @@
...
+        - name: start
+          in: query
+          schema:
+            type: number
+            format: timestamp
+        - name: end
+          in: query
+          schema:
+            type: number
+            format: timestamp
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:166]:
```
type: string
format: instant
```

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 11]
```diff
@@ -0,0 +1,500 @@
...
+          description: All the balance histories from the account
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/BalanceHistory'
+components:
+  schemas:
+    AccountResponse:
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:178]: GetAccountResponse

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 12]
```diff
@@ -0,0 +1,500 @@
...
+      properties:
+        account_id:
+          type: string
+          format: 'uuid'
+          example: '013260eb-7624-40d2-8327-97177a69d8f3'
+        sync_frequency:
+          type: number
+          format: 'uuid'
+          example: 600
+          description: 'How often do we look for new operations'
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:196]: Could you precise the unit (in seconds) ?

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 13]
```diff
@@ -0,0 +1,500 @@
...
+        last_sync_event:
+          $ref: '#/components/schemas/SyncEvent'
+        balance:
+          type: string
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:200]: `format: big integer`
same for all big integer converted to string for json convenience

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 14]
```diff
@@ -0,0 +1,500 @@
...
+          example: 'xpub6BuPWhjLqutPV8SF4RMrrn8c3t7uBZbz4CBbThpbg9GYjqRMncra9mjgSfWSK7uMDz37hhzJ8wvkbDDQQJt6VgwLoszvmPiSBtLA1bPLLSn'
+        scheme:
+          type: string
+          example: 'BIP44'
+        lookahead_size:
+          type: number
+          example: 20
+        network:
+          type: string
+          example: 'MainNet'
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:253]: `enum: [MainNet, TestNet, RegTest]`

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 15]
```diff
@@ -0,0 +1,500 @@
...
+        coin_family:
+          type: integer
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:255]:
```
type: string
enum: [bitcoin]
```

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 16]
```diff
@@ -0,0 +1,500 @@
...
+          example: 'bitcoin'
+        coin:
+          type: string
+          example: 'btc'
```
Comment (nukky94) [inline coins/bitcoin/api/doc/lama-btc-api.yaml:259]: `enum: [btc, btc_testnet, btc_regtest]`

**Review by nukky94 — CHANGES_REQUESTED** [review]

File: coins/bitcoin/api/doc/lama-btc-api.yaml [hunk 17]
```diff
@@ -0,0 +1,500 @@
+openapi: 3.0.1
+info:
+  title: Lama Bitcoin API
+  description: Returns beautifull curated photos
```
[the remark this answers is not in the document]
Comment (jjayet-ledger) [reply coins/bitcoin/api/doc/lama-btc-api.yaml:4]: Ahahah oops

File: coins/bitcoin/api/doc/lama-btc-api.yaml (hunk 1 above)
Comment (jjayet-ledger) [reply coins/bitcoin/api/doc/lama-btc-api.yaml:7]: Oh yeah thx

**Review by nukky94 — APPROVED** [review]

**Review by nukky94 — CHANGES_REQUESTED** [review]
Oh, I didn't see you didn't finish

**Review by nukky94 — APPROVED** [review]

**Pull request merged**